bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/db/EventDaoImpl.kt

```diff
@@ -22,13 +23,33 @@ internal class EventDaoImpl(
   private val eventAdapter = moshi.adapter(Event::class.java)
 
   override fun addEvent(event: Event) {
-    addEventInternal(sqLiteOpenHelper.writableDatabase, event)
+    addEvents(listOf(event))
   }
 
   override fun addEvents(events: List<Event>) {
+    // This approach below is a reference from iOS SDK.
+    // It could be better but This approach will create a minimum impact,
+    // And because the number of pending events in the database is small
+    // So I think we are safe to do this without changing too much
+    val storedEvents = getEvents()
+    val metricsEventUniqueKeys: List<String> = storedEvents.filter {
```

Comment:
This logic should be on the `addEvent` and `addEvents` callers.
Check whether the metrics event should be passed to `addEvent`.

Reply:
Hi @cre8ivejp let me check your suggestion

Reply:
@cre8ivejp  @yshrsmz 
I updated the logic. Please help me to continue reviewing. Thank you.

Reply:
@cre8ivejp @yshrsmz please continue to review this PR. thank you.